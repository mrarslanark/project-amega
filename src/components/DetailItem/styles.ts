import {StyleSheet} from 'react-native';

export const styles = StyleSheet.create({
  wrapper: {
    flex: 1,
    backgroundColor: 'rgba(0, 0, 0, 0.03)',
    paddingVertical: 18,
    paddingHorizontal: 18,
    rowGap: 4,
  },
});
