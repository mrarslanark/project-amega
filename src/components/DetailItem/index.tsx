import React from 'react';
import {View} from 'react-native';
import Text from '../../components/Text';
import {styles} from './styles';

interface DetailItem {
  title: string;
  value?: string;
}

const DetailItem: React.FC<DetailItem> = ({
  title,
  value,
}): React.JSX.Element => {
  return (
    <View style={styles.wrapper}>
      <Text variant="title">{title}</Text>
      <Text variant="medium" numberOfLines={1}>
        {value ?? 'Unkown'}
      </Text>
    </View>
  );
};

export default DetailItem;
