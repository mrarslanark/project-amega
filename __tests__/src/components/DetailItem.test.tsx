import React from 'react';
import {render, screen} from '@testing-library/react-native';
import DetailItem from '../../../src/components/DetailItem';

describe('DetailItem', () => {
  it('should render successfully', () => {
    render(<DetailItem title="Title" value="Value" />);
    expect(screen).toMatchSnapshot();
  });

  it('should render successfully with value to be Unknown', () => {
    render(<DetailItem title="Title" />);

    const textView = screen.getByText('Unknown');
    expect(textView).toHaveTextContent('Unknown');

    expect(screen).toMatchSnapshot();
  });
});
