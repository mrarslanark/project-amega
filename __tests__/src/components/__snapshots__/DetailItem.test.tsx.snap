// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DetailItem.tsx should render successfully 1`] = `
<View
  style={
    {
      "backgroundColor": "rgba(0, 0, 0, 0.03)",
      "flex": 1,
      "paddingHorizontal": 18,
      "paddingVertical": 18,
      "rowGap": 4,
    }
  }
>
  <Text
    style={
      {
        "color": "black",
        "fontSize": 18,
        "fontWeight": "600",
      }
    }
  >
    Title
  </Text>
  <Text
    numberOfLines={1}
    style={
      {
        "color": "black",
        "fontSize": 14,
        "fontWeight": "regular",
      }
    }
  >
    Value
  </Text>
</View>
`;

exports[`DetailItem.tsx should render successfully with value to be Unknown 1`] = `
<View
  style={
    {
      "backgroundColor": "rgba(0, 0, 0, 0.03)",
      "flex": 1,
      "paddingHorizontal": 18,
      "paddingVertical": 18,
      "rowGap": 4,
    }
  }
>
  <Text
    style={
      {
        "color": "black",
        "fontSize": 18,
        "fontWeight": "600",
      }
    }
  >
    Title
  </Text>
  <Text
    numberOfLines={1}
    style={
      {
        "color": "black",
        "fontSize": 14,
        "fontWeight": "regular",
      }
    }
  >
    Unknown
  </Text>
</View>
`;
